text/0001-cdk-update.md

```diff
@@ -0,0 +1,293 @@
+
+#### Preparing Your Stack
+
+In order to ensure that update is not invoked against production stacks, you
+must explicitly designate your stack as being updateable in your application.
+This is a stack-level property, set in the `StackProps` parameter to the stack
+constructor:
+
+```typescript
+const myDevelopmentStack = new ApplicationStack(app, "StackNameForDev", {
+    ...
+    interactiveUpdate: true,
+});
+```
+
+The `interactiveUpdate` property is optional, and when present and `true` the
+`cdk update` functionality will be executed on the stack.
+
+#### Running Update
+
+To run a one-shot update, you can invoke the `cdk update` command on some or all
+of your stacks:
+
+```
+$ cdk update ApplicationStack
+Checking stack ApplicationStack for possible asset update
+ * LambdaFunction[AppFunction]: File asset changed
+ApplicationStack can be updated
+1/1 stacks can be updated:
+Updating LambdaFunction in ApplicationStack
+...
+Update complete!
+```
+
+If the update is to an attribute of the stack that cannot be updated, the
+command will offer to perform a full deployment:
+
+```
+$ cdk update ApplicationStack
+Checking stack ApplicationStack for possible asset update
+ * LambdaFunction[Appfunction]: Function data changed
+ApplicationStack can *not* be rapidly updated.
+Perform a full deployment of ApplicationStack? [Y/n]: Y
+Deploying ApplicationStack
+...
+Done!
+```
+
+When multiple stacks are provided, update only occurs if all pending changes are
+suitable for update. Otherwise, a full deployment is done:
+
+```
+$ cdk update ApplicationStack DataStack
+Checking stack ApplicationStack for possible asset update
+ * LambdaFunction[Appfunction]: Function data changed
+ApplicationStack does not support update.
+Checking stack DataStack for possible asset update
+ * LambdaFunction[StreamFunction]: File asset changed
+DataStack can be updated
+1/2 stacks can be updated: Perform a full deployment? [Y/n]? Y
+Deploying ApplicationStack
+...
+Deploying DataStack
+...
+Done!
+```
+
+In addition to running in a one shot mode, the `cdk update` command also has a
+`--watch` command line option that enable it to monitor the assets on disk and
+perform an update when they change.
+
+```
+$ cdk update --watch ApplicationStack DataStack
+Checking stack ApplicationStack for possible asset update
+ * LambdaFunction[Appfunction]: No Changes
+ApplicationStack can *not* be updated
+Checking stack DataStack for possible asset update
+ * LambdaFunction[StreamFunction]: No Changes
+
+Watching stack inputs for changes:
+ * LambdaFunction[Appfunction]: <spinner>
+ * LambdaFunction[StreamFunction]: <spinner>
+
+```
+
+The watcher can only watch for file changes. For Lambda code, that means that
+directories will be watched for changes and zipped if they change. If the file
+asset is a zip file, then the update will fire whenever that changes. The
+existing docker asset builder will be used to watch for changes in local docker
+images.
+
+#### Resource Support
```

Comment:
I'd like to make sure we also plan to support other things like step function models

Reply:
Agreed. I didn't plan to support them in an initial release, but SFN is a very obvious place where this functionality would be fantastic to have.